### ISSUE TITLE
[Bug]: Previews not working on Xcode 14.X

### ISSUE THREAD

**Issue by knellr** [Appboy/appboy-ios-sdk#349]
### Platform

iOS

### Platform Version

N/A

### Braze SDK Version

4.5.0

### Xcode Version

Xcode 14.X

### Integration Method

Swift Package Manager

### Computer Processor

Apple (M1)

### Repro Rate

100%

### Steps To Reproduce

- Use Swift Package Manager to add AppboyKit as a dependency
- Import AppBoyKit from the app target
- Add a SwiftUI preview to the app target

### Expected Behavior

Preview works as it did on Xcode 13.X

### Actual Incorrect Behavior

HumanReadableSwiftError
SettingsError: noExecutablePath(<IDESwiftPackageStaticLibraryProductBuildable:ObjectIdentifier(0x000060002d064ba0):'AppboyKit'>)

### Additional Information

The issue appears to be discussed here: https://developer.apple.com/forums/thread/707569 and relates to the AppBoyKit product being set to .static explicitly.

We use Swift packages extensively including some of our own modules, but we don't have the setup described in the thread above.

**Comment by giljobs** [comment]
me too 😭 , with same error.

**Comment by hokstuff** [comment]
Hi @knellr,

Looking through the attached thread, it sounds like the issues here are related to how the Previews in Xcode 14 interface with some (static) libraries. Do any of the workarounds in the recent responses [of that thread](https://developer.apple.com/forums/thread/707569) fix the issue for you? There are also occasional responses saying that Xcode 14.1 addressed the issue for them.

**Comment by knellr** [comment]
The issues are still there on Xcode 14.1. I'm not sure what solution you're referring to but I can't see anything from the thread that's relevant to our setup (beyond the suggestions that require updating the SDK's Package.swift) - while we have a variety of packages, including some local, we don't have the structure described in a couple of them - we're importing the SDK straight from our main app project.

**Comment by knellr** [comment]
Ah page 2 wasn't loading properly so I didn't see this one, which looks plausible as it would wrap the static library in a shared library, I'll report back

- Create a local package and add the the intercom dependency to that package
- Remove (SDK) from the main project
- Add the local package as a dependency to the main project

**Comment by knellr** [comment]
I've had a look at the above approach (creating a local package with a dependency on this one) and it doesn't help, whether creating a statically linked library or a framework.

I tried cloning the repository and updating the Package.swift from this repository to remove the "type: .static" definition (as below) and the issue with previews is resolved.

```
  products: [
    .library(name: "AppboyKit", targets: ["AppboyKit"]),
    .library(name: "AppboyUI", targets: ["AppboyUI"]),
    .library(name: "AppboyPushStory", targets: ["AppboyPushStory"])
  ],
```

**Comment by knellr** [comment]
This is fixed in 14.2 RC, as we were probably all hoping :)
